connection.go

```diff
@@ -351,6 +361,16 @@ func Connect(addr string, opts Opts) (conn *Connection, err error) {
 		}
 	}
 
+	if err = checkProtocolVersion(opts.RequiredProtocolVersion, conn.ClientProtocolVersion()); err != nil {
+		return nil, fmt.Errorf("client: %w", err)
+	}
+
+	requiredFeatures := make([]Feature, len(opts.RequiredFeatures))
+	copy(requiredFeatures, opts.RequiredFeatures)
+	if err = checkFeatures(requiredFeatures, conn.ClientFeatures()); err != nil {
+		return nil, fmt.Errorf("client: %w", err)
+	}
```

Comment:
Let's remove the client features check. Still, the idea is to require something from the server.

Reply:
Well, it seems like Tarantool check on both sides (even though it's not really clear through source code):

```
Tarantool 2.11.0-entrypoint-700-gc4b8debee
type 'help' for interactive help
tarantool> box.cfg{listen=3301}
```

```
Tarantool 2.10.0-beta2-283-gb56e012bc
type 'help' for interactive help
tarantool> net_box = require('net.box')
---
...

tarantool> net_box.connect('localhost:3301', {required_protocol_features = {'pagination'}})
---
- peer_protocol_version: 4
  peer_uuid: 7735b4a0-fb8c-4530-afe0-0853888f5342
  opts:
    required_protocol_features:
    - pagination
  host: localhost
  peer_protocol_features:
    transactions: true
    watchers: true
    error_extension: true
    streams: true
  protocol: Binary
  state: error
  error: 'Missing required protocol features: pagination'
  peer_version_id: 133888
  port: '3301'
...
```

(While we were reviewing, new feature had been dropped.)

Reply:
I think `net.box` just can't find `pagination` in an internal mapping. I don't see the point in this in our case: we just limit a user in this case.

Reply:
> I don't see the point in the check in our case: we just limit a user to know some info about server.

I don't like the case when user requires a feature, gets a success and then finds out that we can't really use it. But it seems that, in case of both net.box and a connector, they would fail when they try to use some handles that are not exist (even better in case of Go since it would fail in compile-time). Ok, let's remove it and file a ticket to tarantool/tarantool about inconsistency.